### PR TITLE
Added support for MSE (Massive Storage Engine) in Varnish-Plus

### PR DESCRIPTION
**Pull request by ernestojpg**
##### Summary

The current Varnish collector is only able to collect metrics from the **SMF** (File) and the **SMA** (Malloc) storages. This pull request is adding support for the [**MSE** (Massive Storage Engine)](https://docs.varnish-software.com/varnish-cache-plus/features/mse/) from Varnish-Plus.

I have also renamed some variables/functions to make them more generic, and updated the documentation.

Related to #9657

##### Component Name

Affected component: Varnish collector (python.d plugin)

##### Test Plan

I have tested the change in Varnish-Cache 6.4.0 (Free version) and Varnish-Plus 6.0.6r10 (Commercial/Enterprise version).
Anyway, the change should not break the compatibility with previous versions.

##### Additional Information

I have just added the prefix `MSE` as a valid one for containing Storage metrics.

Example of MSE stats returned in `varnishstat -1`:
```
MSE.mse.c_req                                 0         0.00 Allocator requests
MSE.mse.c_fail                                0         0.00 Allocator failures
MSE.mse.c_fail_malloc                         0         0.00 System allocator failures
MSE.mse.c_bytes                     44791785968  78858778.11 Bytes allocated
MSE.mse.c_freed                     34535437768  60801827.06 Bytes freed
MSE.mse.g_alloc                         1979990          .   Allocations outstanding
MSE.mse.g_bytes                     10256348200          .   Bytes outstanding
MSE.mse.g_space                               0          .   Bytes available
MSE.mse.n_lru_nuked                     6666994     11737.67 Number of LRU nuked objects
MSE.mse.n_lru_moved                           0         0.00 Number of LRU move operations
MSE.mse.n_vary                                0         0.00 Number of Vary header keys
MSE.mse.c_memcache_hit                        0         0.00 Stored objects cache hits
MSE.mse.c_memcache_miss                       0         0.00 Stored objects cache misses
MSE.mse.g_ykey_keys                           0          .   Number of YKeys registered
MSE.mse.c_ykey_purged                         0         0.00 Number of objects purged with YKey
```

Example of the MSE Storage:

Screenshot 2020-12-03 at 22 52 05

### CODE REVIEW

File: collectors/python.d.plugin/varnish/varnish.chart.py [hunk 1]
```diff
@@ -331,14 +337,14 @@ def get_smf_sma_storages(self, server_stats):
         #  ('SMF.', 'ssdStorage.g_smf', '40130'),
         #  ('SMF.', 'ssdStorage.g_smf_frag', '311'),
         #  ('SMF.', 'ssdStorage.g_smf_large', '66')]
-        storages = [name for typ, name, _ in server_stats if typ.startswith(('SMF', 'SMA')) and name.endswith('g_space')]
+        storages = [name for typ, name, _ in server_stats if typ.startswith(('SMF', 'SMA', 'MSE')) and name.endswith('g_space')]
```
Comment (ernestojpg) [inline collectors/python.d.plugin/varnish/varnish.chart.py:340]: This is the main change, adding `MSE` as a valid storage prefix.

File: collectors/python.d.plugin/varnish/README.md [hunk 2]
```diff
@@ -29,16 +30,15 @@ This module produces the following charts:
 -   Backend Connections Statistics in `connections/s`
 -   Requests To The Backend in `requests/s`
 -   ESI Statistics in `problems/s`
--   Memory Usage in `MiB`
```
Comment (ilyam8) [inline collectors/python.d.plugin/varnish/README.md:32]: Why do you delete `Memory Usage` chart from the list?

**Comment by ilyam8** [comment]
@ernestojpg thx for the contribution and for such detailed OP. Very nice job! 👍

File: collectors/python.d.plugin/varnish/README.md (hunk 2 above)
Comment (ernestojpg) [reply collectors/python.d.plugin/varnish/README.md:32]: Upss! It seems this chart does not appear in Varnish-Plus (probably another incompatibility). I will take a look and create another pull request for fixing that. Let me put this chart back in the documentation. Thanks!

Comment (ernestojpg) [reply collectors/python.d.plugin/varnish/README.md]: Done. I put the chart back in the list.

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
Thank you @ernestojpg !

**Comment by OdysLam** [comment]
Welcome to our community @ernestojpg and thank you for your contribution indeed :) 

In the meantime, if you are interested, we have an awesome community at https://community.netdata.cloud. We have all sorts of discussions about netdata and monitoring :) 

Cheers!

**Pull request merged**